chia/data_layer/data_layer_wallet.py

```diff
@@ -138,34 +135,26 @@ def id(self) -> uint32:
         return self.wallet_info.id
 
     @classmethod
-    async def create_new_dl_wallet(
-        cls: Type[_T_DataLayerWallet],
-        wallet_state_manager: WalletStateManager,
-        wallet: Wallet,
-        name: Optional[str] = "DataLayer Wallet",
-    ) -> _T_DataLayerWallet:
+    async def create_or_get(cls, wallet_state_manager: WalletStateManager) -> DataLayerWallet:
         """
         This must be called under the wallet state manager lock
         """
 
-        self = cls()
-        self.wallet_state_manager = wallet_state_manager
-        self.log = logging.getLogger(name if name else __name__)
-        self.standard_wallet = wallet
-
-        for _, w in self.wallet_state_manager.wallets.items():
-            if w.type() == uint8(WalletType.DATA_LAYER):
-                raise ValueError("DataLayer Wallet already exists for this key")
-
-        assert name is not None
-        self.wallet_info = await wallet_state_manager.user_store.create_wallet(
-            name,
-            WalletType.DATA_LAYER.value,
-            "",
-        )
-        self.wallet_id = uint8(self.wallet_info.id)
-
-        await self.wallet_state_manager.add_new_wallet(self, self.wallet_info.id)
+        for wallet in wallet_state_manager.wallets.values():
+            if wallet.type() == uint8(WalletType.DATA_LAYER):
+                assert isinstance(
+                    wallet, DataLayerWallet
+                ), f"WalletType.DATA_LAYER should be a DataLayerWallet instance got: {type(wallet).__name__}"
+                return wallet
+
+        async with wallet_state_manager.lock:
```

Comment:
the lock is new, right? and looks like it's fixing possible races. If any `async` call is introduced above though, we'd have a problem. It looks like it would be safer to acquire the lock before looking for an existing wallet.

e.g. if any call is made to yield while looking for the existing wallet, another task can create it without us noticing, and we then create it again